### DISCUSSION TITLE
New instance of TextCleanerScript

### DISCUSSION THREAD

**Discussion by fufurax** [dlemstra/FredsImageMagickScripts.NET#32] [Q&A]
Hi,
For personnal project i try to use your project to clean part of picture.

  ```
TextCleanerScript<ushort> cleaner = new TextCleanerScript<ushort>(new MagickFactory());
var i = cleaner.Execute(ImageScreen);
if (_DebugSavePics) i.ToBitmap().Save($"ScreenCapture_Cleaner_crop_ocr.png");

```
it returns a blank (full white) image.

1st step : https://ibb.co/LdPNLQS      >> needed data the %
2nd step after crop : https://ibb.co/Bw116xd   > small % on previous top right (in my previous step it's 2 %) 
3 step : https://ibb.co/cyJZfPj     > as you can see there is small points

My goal is to clean image before sending to tesseract.

Thanks for your help

**Comment by fmw42** [comment]
Textcleaner is use for black text on white background, not text on a picture background.  See examples at http://www.fmwconcepts.com/imagemagick/textcleaner/index.php

**Comment by fufurax** [comment]
> Textcleaner is use for black text on white background, not text on a picture background. See examples at http://www.fmwconcepts.com/imagemagick/textcleaner/index.php

Look to my picture step 3 ;)

**Comment by fmw42** [comment]
Your image is not compatible with textcleaner.

**Comment by fufurax** [comment]
> Your image is not compatible with textcleaner.

Ok but why ?

**Comment by fmw42** [comment]
The text needs to be black on a whitish background, not on a pictures or graphic background

**Comment by fufurax** [comment]
So like step 3...see first message. 

3 step : https://ibb.co/cyJZfPj

**Comment by fmw42** [comment]
Sorry, I do not understand your question nor your steps.  I am not Magick.Net user. So you may have to explain better and then perhaps the Magick.Net developer and I may be able to understand what you are doing.

**Comment by fufurax** [comment]
https://github.com/ImageMagick/ImageMagick/discussions/5660

my process 

1. take screen shot (1366 * 768) : https://ibb.co/Hx5PZDL
2. crop (57*72) : https://ibb.co/K2YfVGT
3. "cleaning colors" (57*72) : https://ibb.co/dfj5MTD
```
 ImageScreen.ColorFuzz = new Percentage(10);
                    ImageScreen.Transparent(_Zcolors[0]);
                    ImageScreen.Transparent(_Zcolors[1]);
                    ImageScreen.Transparent(_Zcolors[2]);
                    ImageScreen.Transparent(_Zcolors[3]);
                    ImageScreen.Alpha(AlphaOption.Extract);
                    ImageScreen.Negate();

                    ImageScreen.Transparent(MagickColors.Black);
                    ImageScreen.Alpha(AlphaOption.Extract);
                    ImageScreen.Negate();
                    ImageScreen.Threshold(new Percentage(15));
```
4. compose to 300*300 >> https://ibb.co/mSwzD6L
            ```
MagickImage Overlay = new MagickImage("xc:white", 300, 300);
            Overlay.Density = Main.Density;
            Overlay.Transparent(MagickColors.White);
            Overlay.Composite(Main, CompositeOperator.SrcOver);

```
5.  and send it to tesseract. 

I can't do more...

**Comment by fmw42** [comment]
You can clean up the dots using mophology close.
